### PR TITLE
Raise libheif version requirement to at least 1.13.0

### LINKED ISSUE #13306: Wrong orientation and dimensions with iPhone HEIC pictures
**Did you buy darktable from an application store ?**

No

**Describe the bug/issue**

I have Darktable 4.2.0 from flathub (https://github.com/flathub/org.darktable.Darktable; runtime is org.gnome.Platform/x86_64/43 released 2022-12-28 00:10:05 +0000 from commit 2e472987d6fb03324dcc9b19e75ee96b3a87918dd8649f9eb64009d6d9007ea4)

Orientation of HEIC files is off (with the orientation module enabled, which is by default).  It seems like rotations are applied twice.  In most cases, the image is also tiled: a square section of the image is correct, but the outer dimensions are swapped, and the image repeats itself to fill.

Orientation is set both with tags from the EXIF group and the QuickTime group, for example:

```
exiftool -groupNames 67575746302__5F8F44D1-2185-469F-A7D1-74DD703B204B.heic |grep -Ei 'orient|rotat'
[QuickTime]     Rotation                        : 270
[EXIF]          Orientation                     : Rotate 90 CW
```

Disabling the orientation module makes the orientation correct, but does not fix the tiling issue for images that are off by ±90°, so it's only a partial workaround.

**To Reproduce**
Add the attached images to Darktable 4.2.0 and display them.
Toggle the orientation module on and off.

**Expected behavior**
Show images with the correct orientation when the orientation module is enabled.
Don't apply any orientation (discarding both kinds of EXIF metadata) when the orientation module is disabled.

**Screenshots**

screenshot 1
screenshot 2

** Attachments **

[iphone-heic-trouble.zip](https://github.com/darktable-org/darktable/files/10373607/iphone-heic-trouble.zip)

**Platform**

* darktable version : 4.2.0
* Linux - Distro    : Ubuntu 22.10

**Additional context**

 - Are the steps above reproducible with a fresh edit (i.e. after discarding history)? **yes**
 - Is the issue still present using an empty/new config-dir (e.g. start darktable with --configdir "/tmp")? **My config is almost default, except I use the modern chromatic adaptation which enables orientation by default**
 - Do you use lua scripts?
**no**

I found a few related past issues with HEIF orientation and iPhone photos: https://github.com/darktable-org/darktable/issues/10906 https://github.com/darktable-org/darktable/issues/13045

### PR DESCRIPTION
**Pull request by g2p**
This fixes a bug that breaks orientation processing and crops everything but a square subset of the image.

Fixes #13306.

### CODE REVIEW

**Review by TurboGit — COMMENTED** [review]
I suppose this is ok for 4.2.1 ?

**Comment by TurboGit** [comment]
@g2p : Do you have an idea how widely available is this version?

**Comment by kmilos** [comment]
https://repology.org/project/libheif/versions

**Comment by TurboGit** [comment]
@kmilos : I didn't know the repology service! Nice, thanks.

It is on Debian-12, homebrew, Unbuntu 22/10, MSYS2... Looks good to me for an option deps. Agreed?

**Comment by g2p** [comment]
Sure!  I'd also note that stable distros are unlikely to update Darktable in isolation.  Packagers can patch the version check if they are making a deliberate choice, but they need the requirement to have a working version.

**Comment by victoryforce** [comment]
> It is on Debian-12, homebrew, Unbuntu 22/10, MSYS2... Looks good to me for an option deps. Agreed?

It looks like it's getting harder and harder to build a fully functional dt on anything older than the latest distro releases. IMHO, it's time to offer AppImage as a convenient enough solution. Currently, it is quite realistic to provide an official AppImage build that will work on Ubuntu 20.04+, Fedora 32+, Debian 11+. I will take care of it this week.

**Comment by kmilos** [comment]
> It looks like it's getting harder and harder to build a fully functional dt on anything older than the latest distro releases.

I'm personally in favour of having a big fat warning in CMake (an known issues in release notes or readme) for the 4.2 branch, and only a hard requirement/error on master.

**Comment by parafin** [comment]
I don't think HEIF support is that big of a feature to be missed by a lot of people if not available in some distro build. It is optional after all for a reason.

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -400,7 +400,7 @@ if(USE_HEIF)
   if(NOT TARGET heif)
     find_package(libheif MODULE)
```
Comment (kmilos) [inline src/CMakeLists.txt:401]: add 1.13.0 for the "module" search, it'll give a more meaningful message in the log

(doesn't work for the "config" search above as it looks for exact version match, and not for greater or equal...)

**Comment by TurboGit** [comment]
I'm not for a big warning, HEIF is not a main stream format and just optional for dt. Let's keep things simple.

**Comment by Nilvus** [comment]
I agree, HEIF is not used by a lot of people and so should remain optional dep. Anyway, @victoryforce proposal to add an AppImage is a good idea to complement actual packages provided.

File: src/CMakeLists.txt (hunk 1 above)
Comment (TurboGit) [reply src/CMakeLists.txt:401]: @g2p : Just a ping in case you missed this comment from @kmilos. I'll then merge for 4.2. TIA.

Comment (g2p) [reply src/CMakeLists.txt]: Like this?  I don't know cmake much.

Comment (kmilos) [reply src/CMakeLists.txt]: Yep, good to go!

**Review by TurboGit — APPROVED** [review]
Thanks!

**Comment by TurboGit** [comment]
Merged in master & 4.2 branch.

**Pull request merged**